pinot-core/src/main/java/com/linkedin/pinot/core/io/compression/PassThroughDecompressor.java

```diff
@@ -26,9 +25,11 @@
  */
 public class PassThroughDecompressor implements ChunkDecompressor {
   @Override
-  public int decompress(ByteBuffer inCompressed, ByteBuffer outDecompressed)
-      throws IOException {
+  public int decompress(ByteBuffer inCompressed, ByteBuffer outDecompressed) {
     outDecompressed.put(inCompressed);
-    return inCompressed.limit();
+
+    // Flip the output ByteBuffer for reading.
```

Comment:
any test case that would fail without this fix?

Reply:
Yes, the testcase modified in the PR will fail if we remove the fix.